src/crypto/index.ts

```diff
@@ -3591,25 +3596,27 @@ export class Crypto extends TypedEventEmitter<CryptoEvent, CryptoEventHandlerMap
             return;
         }
 
-        // check when we last forced a new session with this device: if we've already done so
+        // check when we can force a new session with this device: if we've already done so
         // recently, don't do it again.
-        const lastNewSessionDevices = this.lastNewSessionForced.getOrCreate(sender);
-        const lastNewSessionForced = lastNewSessionDevices.getOrCreate(deviceKey);
-        if (lastNewSessionForced + MIN_FORCE_SESSION_INTERVAL_MS > Date.now()) {
+        const forceNewSessionRetryTimeDevices = this.forceNewSessionRetryTime.getOrCreate(sender);
+        const forceNewSessionRetryTime = forceNewSessionRetryTimeDevices.getOrCreate(deviceKey);
+        if (forceNewSessionRetryTime > Date.now()) {
             logger.debug(
                 "New session already forced with device " +
                     sender +
                     ":" +
                     deviceKey +
-                    " at " +
-                    lastNewSessionForced +
-                    ": not forcing another",
+                    ": not forcing another until at least ",
+                    forceNewSessionRetryTime
```

Comment:
suggest updating this to backtick interpolation while you are here

```
`New session already forced with device ${sender} ...`
```

etc

Reply:
also suggest `new Date(forceNewSessionRetryTime).toUTCString()` to make the timestamp readable